test/test_foreach.py

```diff
@@ -179,6 +199,17 @@ def test_binary_op(self, device, dtype, op, is_fastpath):
                     self.assertEqual([t.grad for t in tensors], [t.grad for t in ref_tensors])
                     if isinstance(rhs_arg, list) and isinstance(rhs_arg[0], torch.Tensor):
                         self.assertEqual([t.grad for t in rhs_arg], [t.grad for t in ref_rhs_arg])
+                    tensors = [1 * t.clone().detach().requires_grad_() for t in tensors]
+                    ref_tensors = [1 * t.clone().detach().requires_grad_() for t in tensors]
+                    inplace_op([tensors, rhs_arg], is_cuda=False, is_fastpath=False, zero_size=zero_size)
+                    assert_multiple_grad_fns(tensors, self)
+
+                    inplace_ref([ref_tensors, rhs_arg])
+                    torch.autograd.backward(sum([t.clone() for t in ref_tensors]).sum(), inputs=ref_tensors)
+                    if inplace_op.func not in (torch._foreach_minimum_, torch._foreach_maximum_):
+                        torch.autograd.backward(sum([t.clone() for t in tensors]).sum(), inputs=tensors)
+                        grads = [t.grad for t in tensors]
+                        self.assertEqual(grads, [t.grad for t in ref_tensors])
```

Comment:
Do we have a test where not all inputs tensors require grad?
- We should make sure that that case does not error and the behavior is as-if we did a for loop, i.e. only the outputs corresponding to inputs that require grad should require grad. 
- We should also make sure that only inputs (or outputs that correspond to inputs) that require grad should be saved for backward.

Reply:
I haven't added that kind of test. what do you think about the test 4941cd22b297a5e1d81221bccaed93aba4d4d8ec added?

Reply:
Look ok to me! (if we wanted to be more exhaustive we could test more combinations, but this is also fine) btw, is it named "test_inplace_foreach_multi_grad_fn" instead of something specific like "test_inplace_foreach_preserves_requires_grad" because we plan to add more checks there? (I think that would be a good idea :)